src/main/scala/com/lifeomic/fhirlib/v3/resources/Patient.scala

```diff
@@ -64,41 +64,34 @@ class Patient(override val id: Option[String],
 
   def getRaceCoding(urlContains: String = "us-core-race",
                     systemContains: String = "Race"): Option[Coding] = {
-    try {
-      val ext = extension.get.filter(_.url.getOrElse("") contains urlContains).headOption
-      val coding = ext.get.valueCodeableConcept.get.coding.get
-      return coding.filter(x => x.system.nonEmpty && (x.system.get contains systemContains)).headOption
-    } catch {
-      case _: Throwable => None
-    }
+    getExtensionCoding(urlContains, Some(systemContains))
   }
 
   def getEthnicityCoding(urlContains: String = "us-core-ethnicity",
                          systemContains: String = "Ethnicity"): Option[Coding] = {
-    try {
-      val ext = extension.get.filter(_.url.getOrElse("") contains urlContains).headOption
-      val coding = ext.get.valueCodeableConcept.get.coding.get
-      return coding.filter(x => x.system.nonEmpty && (x.system.get contains systemContains)).headOption
-    } catch {
-      case _: Throwable => None
-    }
+    getExtensionCoding(urlContains, Some(systemContains))
   }
 
-  def getLanguageCodes(): List[String] = {
-    val langs = ListBuffer[String]()
+  def getLanguageCodings(): List[Coding] = {
     if (communication.isEmpty) {
-      return langs.toList
+      return List[Coding]()
     }
+
+    val codings = ListBuffer[Coding]()
     communication.get.foreach(comm => {
-      if (!comm.language.coding.isEmpty) {
+      if (comm.language.coding.nonEmpty) {
         comm.language.coding.get.foreach(coding => {
           if (!coding.code.isEmpty) {
-            langs += coding.code.get
+            codings += coding
           }
         })
       }
     })
-    return langs.toList
+    return codings.toList
+  }
+
+  def getLanguageCodes(): List[String] = {
+    return getLanguageCodings().map(_.code.get)
```

Comment:
code might be null and so `code.get` would throw an exception. Can't make any assumptions. Instead you could do a flatmap, I think. Something like this should work:
```
getLanguageCodings().flatMap(x => {
    if (x.code.nonEmpty) {
       List(x.code.get)
    } else {
        List()
    }
})
```

Reply:
Should be ok, we are checking if `code` is empty before adding to the list of codings in `getLanguageCodings`